qokit/qaoa_circuit.py

```diff
@@ -52,12 +51,13 @@ def append_x_term(qc: QuantumCircuit, q1: int, beta: float) -> None:
 def append_cost_operator_circuit(qc: QuantumCircuit, terms: Sequence, gamma: float) -> None:
     for term in terms:
         if len(term) == 2 and isinstance(term[1], tuple):
-            coeff, (i, j) = term
-            append_z_prod_term(qc, (i, j), gamma * coeff / 2)
+            coeff, term_tuple = term
+            append_z_prod_term(qc, term_tuple, gamma * coeff / 2)
```

Comment:
Also a question about the old code, why do terms with coefficients differ from terms without by a factor of 2? Which problem relies on this convention?

Reply:
I think there should have been a factor of 2 in both cases. I included this missing factor of 2 in the recent commit.

Reply:
Could you confirm this is indeed the case? I find it weird that we divide by 2 before calling `append_z_prod_term()` and immediately multiply it back in `append_z_prod_term()`.

Reply:
As discussed, the factor of 2 remains there in order to keep the convention (used in QOKit) of multiplying `gamma` and `beta` when applying gates in cost and mixer terms. Added a docstring to clarify this issue.